apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/RetryHelpers.cs

```diff
@@ -28,8 +28,14 @@ namespace Google.Cloud.Spanner.Data.CommonTesting
     public static class RetryHelpers
     {
         // Allow up to 10 seconds each time.
-        private static readonly BackoffSettings s_backoffSettings =
-            new BackoffSettings(delay: TimeSpan.FromMilliseconds(50), maxDelay: TimeSpan.FromSeconds(1), delayMultiplier: 2);
+        private static readonly RetrySettings s_retrySettings =
+            RetrySettings.FromExponentialBackoff(
+                maxAttempts: int.MaxValue,
+                initialBackoff: TimeSpan.FromMilliseconds(50),
+                maxBackoff: TimeSpan.FromSeconds(1),
+                backoffMultiplier: 2,
+                retryFilter: ignored => false,
```

Comment:
For the possible changes in RetrySettings on GAX, maybe consider having RetrySettings<TException> so that the filter predicate can be on any exception and not just RpcException?

Reply:
I meant generic RetrySettings on the filter exception type. I originally wrote RetrySettings "lt"TException"gt" but markup interpreted that and didn't show the generic part :)

Reply:
Hmm. Not sure, but let's consider that separately in GAX. I suspect it would be messy to have a generic RetrySettings everywhere.